src/js/index.js

```diff
@@ -21,15 +21,74 @@ const allowDifficultySelect = shouldAllow => sortingInput.querySelectorAll('.dif
 		node.disabled = !shouldAllow;
 	});
 
+const parameters = {
+	tags: {
+		default: '',
+		getValue: () => selectr.getValue().join(' '),
+		setValue: value => selectr.setValue(value.split(' '))
+	},
+	difficulty: {
+		default: 'all',
+		getValue: () => filterInput.value,
+		setValue: value => {
+			filterInput.value = ['easy', 'medium', 'hard'].includes(value) ?
+				value :
+				'all';
+		}
+	},
+	sort: {
+		default: 'alphabetical',
+		getValue: () => sortParams.sortMethod.toLowerCase(),
+		setValue: value => {
+			sortParams.sortMethod = ['alphabetical', 'difficulty'].includes(value) ?
+				value :
+				'alphabetical';
+		}
+	},
+	order: {
+		default: 'asc',
+		getValue: () => `${sortParams.order.split('SC')[0]}sc`.toLowerCase(),
+		setValue: value => {
+			sortParams.order = value === 'desc' ?
+				'descending' :
+				'ascending';
+		}
+	}
+};
+
 let search;
 let selectr;
+const sortParams = {
+	sortMethod: sortingInput.value.split('_')[0],
+	order: sortingInput.value.split('_')[1]
+};
+
+function updateUrl() {
+	const newSearch = new URLSearchParams(window.location.search);
+	for (const parameter in parameters) {
+		if (Object.prototype.hasOwnProperty.call(parameters, parameter)) {
+			const paramValue = parameters[parameter].getValue();
+			if (paramValue === '') {
```

Comment:
It misses the part I added to remote values that are set to default from the SearchString.

```suggestion
			if (['', parameters[parameter].default].includes(paramValue)) {
```